src/ui/QueryForCommerce/QueryForCommerce.ts

```diff
@@ -0,0 +1,71 @@
+import { Component } from '../Base/Component';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { Assert } from '../../misc/Assert';
+import { QueryEvents, IBuildingQueryEventArgs } from '../../events/QueryEvents';
+import { AnalyticsEvents, IChangeAnalyticsCustomDataEventArgs } from '../../events/AnalyticsEvents';
+import { Initialization } from '../Base/Initialization';
+import { exportGlobally } from '../../GlobalExports';
+
+export interface IQueryForCommerceOptions {
+  listing?: string;
```

Comment:
do we want to add the  catalogId in this component as well?

Reply:
Yes we should definitely add it, but I would keep it for another PR and keep the scope of this one like it is right now.